codec-http/src/main/java/io/netty/handler/codec/http/websocketx/extensions/WebSocketServerExtensionHandler.java

```diff
@@ -143,9 +142,9 @@ private void handlePotentialUpgrade(final ChannelHandlerContext ctx,
                     headers.set(HttpHeaderNames.SEC_WEBSOCKET_EXTENSIONS, newHeaderValue);
                 }
             }
-            promise.addListener((ChannelFutureListener) future -> {
+            promise.addListener(future -> {
                 if (future.isSuccess()) {
-                    ctx.pipeline().remove(WebSocketServerExtensionHandler.this);
+                    ctx.pipeline().remove(this);
```

Comment:
I am surprised this works :)

Reply:
Lambdas don't have their own `this`, so now that this is no longer an anonymous inner class, we don't have to qualify the `this` reference.

Reply:
TIL!